benchmark/profile_restful_api.py

```diff
@@ -66,21 +66,28 @@ def __init__(self,
                  temperature: float = 0.8,
                  top_p: float = 1.0,
                  csv: str = '',
+                 api_key: Optional[str] = None,
+                 model_name: Optional[str] = None,
                  **kwargs):
         self.tokenizer = Tokenizer(tokenzier_path)
         self.server_addr = server_addr
         self.temperature = temperature
         self.top_p = top_p
         self.csv = csv
-        client = APIClient(self.server_addr)
-        self.model_name = client.available_models[0]
+        self.api_key = api_key
+        client = APIClient(self.server_addr, api_key=self.api_key)
```

Comment:
What's the purpose of adding the `api_key` in the benchmark script, and in what scenarios would it be used?

Reply:
1. To test the service speed from provider such as openai, some cloud.
2. Users often have only one set of environmen due to lack of resources, which needs a api_key to access.

Reply:
> 1. To test the service speed from provider such as openai, some cloud.
> 2. Users often have only one set of environmen due to lack of resources, which needs a api_key to access.

make sense